app/waf/src/repositories/common.ts

```diff
@@ -22,4 +31,42 @@ export class CommonRepository<
 
     return super.updateById(id, data, options);
   }
+
+  findEntitiesByTenant(tenantId: string): Promise<T[]> {
+    const filter: Filter = {where: {tenantId: tenantId}};
+    // maybe remove this "magic conversion" someday.
+    return this.find(filter as (Filter<T>));
+  }
+
+  async findEntityByTenant(tenantId: string, id: ID): Promise<T> {
+    // have to do this, since findById filter does not work at all.
+    const filter: Filter = {where: {and: [{tenantId: tenantId}, {id: id}]}};
+
+    let data = await this.find(filter as (Filter<T>));
+    if (data.length !== 0) {
```

Comment:
How about 
```
if (data.length !== 1) {
      throw new EntityNotFoundError(this.entityClass.name, id);
    } else {
      this.logger.debug('find resource ' + data[0]);
      // return the first one we found.
      return data[0];
    }
```
Anyway, your implementation is just OK.

Reply:
OK, if everyone happy to change, I can change it, hmm... but `data.length !== 1` it does not really mean  we do not find any resource/entity....